ImageAnalysis/image_analysis/base.py

```diff
@@ -122,6 +122,76 @@ def apply_config(self, config_file):
+            Parameters
+            ----------
+            return_image : 2d array
+                Image to be returned to labview.  Will be converted to UInt16
+            return_scalars : dict
+                Dictionary of scalars from python analysis.  To be passed back to labview correctly, the keys for each
+                entry need to match those given in labview_adapters.json for this analyzer class
+            return_lineouts : list(np.ndarray)
+                Lineouts to be returned to labview.  Need to be given as a list of 1d arrays (numpy or otherwise)
+                If not given, will return a 1x1 array of zeros.  If in an incorrect format, will return a 1x1 array of
+                zeros and print a reminder message.  If the arrays in the list are of unequal length, all arrays get
+                padded with zeros to the size of the largest array.  Also, will be returned as a 'float64'
+            input_parameters : dict
+                Dictionary of the input parameters given to the analyzer.  If none is given, will call the class's
+                self.build_input_parameter_dictionary() function to generate one from the class variables.  This is not
+                returned to Labview so it can contain anything one might find useful in post-analysis
+
+            Returns
+            -------
+            return_dictionary : dict
+                Dictionary with the correctly formatted returns that labview adapters is expecting.
+                "analyzer_input_parameters": input_parameters
+                "analyzer_return_dictionary": return_scalars
+                "processed_image_uint16": uint_image
+                "analyzer_return_lineouts": return_lineouts
+
+            """
+        uint_image = return_image.astype(np.uint16)
+
+        if return_scalars is None:
+            return_scalars = dict()
+        elif not isinstance(return_scalars, dict):
+            print("return_scalars must be passed as a dict!")
+            return_scalars = dict()
+
+        if isinstance(return_lineouts, np.ndarray) and return_lineouts.ndim == 2:
+            return_lineouts = return_lineouts.astype(np.float64)
+        else:
+            if return_lineouts is not None:
+                if not isinstance(return_lineouts, list):
+                    print("return_lineouts must be passed as a list of 1d arrays!")
+                    return_lineouts = None
+                else:
+                    for lineout in return_lineouts:
+                        shape = np.shape(lineout)
+                        if len(shape) != 1 or shape[0] < 2:
+                            print("return_lineouts must be passed as a list of 1d arrays!")
+                            return_lineouts = None
+                            break
+            if return_lineouts is None:
+                return_lineouts = np.zeros((1, 1), dtype=np.float64)
+            else:
+                max_length = max(len(lineout) for lineout in return_lineouts)
```

Comment:
Nice use of generator instead of list! If you want to get extra-fancy, you can write: `max(map(len, return_lineouts))`

Reply:
Ooh nice, I like that!  There's always a clever way to remove `for` usages it seems